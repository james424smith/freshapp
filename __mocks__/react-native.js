import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
