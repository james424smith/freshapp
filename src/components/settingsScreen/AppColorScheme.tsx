import React from "react";
import { View, TouchableOpacity } from "react-native";
import styles from "../../styles/more_screens/AppColorSchemeStyles";
import colors from "../../constants/styles/colors";
import Text from "../StyledText";
import { Icon } from "react-native-elements";
import { useTheme } from "@react-navigation/native";
import { IRootReducerType } from "../../redux/reducers";
import { useSelector } from "react-redux";
type Props = {
  changeModalVisible: () => void;
};
const AppColorScheme = (props: Props) => {
  const { changeModalVisible } = props;
  const theme = useTheme();

  const selectedIndex = useSelector<IRootReducerType, string>(
    ({ darkModeOptionsReducer }) => darkModeOptionsReducer.darkModeOptionsValue
  );

  return (
    <TouchableOpacity
      testID={"change-theme"}
      style={styles.mainStyle}
      onPress={changeModalVisible}
    >
      <Icon
        name="brightness-2"
        size={25}
        color={colors.white}
        style={styles.iconStyle}
      />
      <View style={styles.content}>
        <View>
          <Text style={{ ...styles.text, color: theme.colors.text }}>
            Dark Mode
          </Text>
        </View>
        <View style={styles.contentContainer}>
          <Text style={[styles.text, { color: colors.darkInputText }]}>
            {selectedIndex}
          </Text>
          <Icon name="chevron-right" color={colors.darkInputText} />
        </View>
      </View>
    </TouchableOpacity>
  );
};

export default AppColorScheme;
