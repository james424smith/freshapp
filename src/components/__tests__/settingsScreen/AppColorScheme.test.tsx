import React from "react";
import AppColorScheme from "../../settingsScreen/AppColorScheme";
import { render, cleanup } from "@testing-library/react-native";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";

describe("AppColorScheme snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it("should render correct ui", () => {
    const changeModalVisible = jest.fn();
    const mockedState = {
      darkModeOptionsReducer: {
        darkModeOptionsValue: "dark",
      },
    };
    const mockedStore = configureMockStore();
    const { toJSON } = render(
      <Provider store={mockedStore(mockedState)}>
        <AppColorScheme changeModalVisible={changeModalVisible} />
      </Provider>
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
