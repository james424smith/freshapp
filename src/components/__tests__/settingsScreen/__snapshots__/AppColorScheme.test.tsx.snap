// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppColorScheme snapshot test should render correct ui 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "stretch",
      "flexDirection": "row",
      "marginTop": 20,
      "opacity": 1,
      "paddingBottom": 15,
      "paddingTop": 15,
    }
  }
  testID="change-theme"
>
  <View
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#78849E",
          "borderRadius": 20,
          "height": 40,
          "justifyContent": "center",
          "width": 40,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 25,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#C5C5C5",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginLeft": 10,
      }
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "color": "rgb(28, 28, 30)",
              "fontSize": 18,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Dark Mode
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 18,
                "fontWeight": "bold",
              },
              Object {
                "color": "#8E9397",
              },
            ],
          ]
        }
      >
        dark
      </Text>
      <View
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#8E9397",
                    "fontSize": 24,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
